uwblueprint/ccbc
# Admin Authentication: Create Account + Setup Password email

## Notion ticket link

This PR combines both of these tickets:
[Admin Authentication: Create Account](https://www.notion.so/uwblueprintexecs/Admin-Authentication-Create-Account-b2b23688973543458dc621ada42fc469)
[Admin Authentication: Setup Password email](https://www.notion.so/uwblueprintexecs/Admin-Authentication-Setup-Password-email-ffb78a0866aa48c4adb61ec24b7ac2ab)



## Implementation description
* Split the signup flow into 2 pages: 1. verify access token 2. set new password
* Created new URL for verify-user mode which is sent in the email to the user
* After the user enters the correct access code, they are redirected to new URL for set-password mode
* Added conditional rendering on pages to deal with link expiring and other error states.



## Steps to test
1. Ensure you don't have an account set up already by clearing your local storage and checking the firebase authentication console to make sure your email id is not there. 
2. Go to http://localhost:3000/signup
3. Enter the details of the account you want invite and click send (ignore the redirect to the home page, take note that your local storage did not change)
4. Go to your email and click on the link (if you get an error saying the site can't be reacher, changed the undefined in the URL to localhost:3000 -> this is because you need to pull the latest secrets for your .env file)
5. Enter your access code and go through the Verification and Setup Password pages -> you should now be logged in 
6. After you have set up your password, you should not be able to go to the set up password page again (to check this, clear the local storage then try to access that link from your email again)



## What should reviewers focus on?
* signInWithEmailAndPassword() in VerifyAccessCode.tsx creates a 400 error in console, we need to fix that
* What should I do if there are issues with logging the user in initially? (Check try catch in useEffect in VerifyAccessCode.tsx file -> should we show some kind of error page for this?)


## Checklist
